src/main/java/com/ita/if103java/ims/service/impl/ItemServiceImpl.java

```diff
@@ -32,41 +31,44 @@
     private SavedItemDao savedItemDao;
     private WarehouseDao warehouseDao;
     private WarehouseDtoMapper warehouseDtoMapper;
+    private TransactionDao transactionDao;
+    private EventService eventService;
 
     @Autowired
-    public ItemServiceImpl(ItemDtoMapper itemDtoMapper, SavedItemDtoMapper savedItemDtoMapper, ItemDao itemDao,
-                           SavedItemDao savedItemDao, WarehouseDao warehouseDao,
-                           WarehouseDtoMapper warehouseDtoMapper) {
+    public ItemServiceImpl(ItemDtoMapper itemDtoMapper, SavedItemDtoMapper savedItemDtoMapper, ItemDao itemDao, SavedItemDao savedItemDao, WarehouseDao warehouseDao, WarehouseDtoMapper warehouseDtoMapper, TransactionDao transactionDao, EventService eventService) {
         this.itemDtoMapper = itemDtoMapper;
         this.savedItemDtoMapper = savedItemDtoMapper;
         this.itemDao = itemDao;
         this.savedItemDao = savedItemDao;
         this.warehouseDao = warehouseDao;
         this.warehouseDtoMapper = warehouseDtoMapper;
+        this.transactionDao = transactionDao;
+        this.eventService = eventService;
     }
 
-
     @Override
-    public List<ItemDto> findItems() {
-        List<Item> items = itemDao.getItems();
-        items.sort(Comparator.comparing(Item::getName));
-        return itemDtoMapper.toDtoList(items);
-
+    public List<ItemDto> findSortedItem(Pageable pageable) {
+        String sort = pageable.getSort().toString().replaceAll(": ", " ");
+        return itemDtoMapper.toDtoList(itemDao.getItems(sort, pageable.getPageSize(), pageable.getOffset()));
     }
 
     @Override
     public ItemDto findById(Long id) {
         return itemDtoMapper.toDto(itemDao.findItemById(id));
+
     }
 
     @Override
-    public SavedItemDto addSavedItem(SavedItemDto savedItemDto) {
-        SavedItem savedItem = savedItemDtoMapper.convertSavedItemDtoToSavedItem(savedItemDto);
+    public SavedItemDto addSavedItem(SavedItemDto savedItemDto, User user, Associate associate) {
+        SavedItem savedItem = savedItemDtoMapper.toEntity(savedItemDto);
         savedItem.setItemId(itemDao.findItemByName(savedItemDto.getItemDto().getName()).getId());
         if (isEnoughCapacityInWarehouse(savedItemDto)) {
             savedItemDao.addSavedItem(savedItem);
+            transactionDao.create(ItemTransactionUtil.createTransaction(savedItemDto, user, associate, TransactionType.IN));
+            eventService.create(ItemEventUtil.createEvent(savedItemDto, user, EventName.ITEM_CAME));
             return savedItemDto;
         }
+        eventService.create(ItemEventUtil.createEvent(savedItemDto, user, EventName.LOW_SPACE_IN_WAREHOUSE));
```

Comment:
**IMPORTANT: In order to not block your PR please do this in next one!**
This event should be logged when warehouse is loaded more than 90 %. So need to add additional condition